plugins/org.jboss.tools.ws.jaxrs.ui/src/org/jboss/tools/ws/jaxrs/ui/wizards/JaxrsElementCreationUtils.java

```diff
@@ -221,10 +224,10 @@ public static IPackageFragment getProjectTopLevelPackage(final IPackageFragmentR
 			currentFolder = (IFolder) currentFolder.members()[0];
 		}
 		if(currentFolder != null) {
-			return (IPackageFragment) packageFragmentRoot.getJavaProject().findPackageFragment(currentFolder.getFullPath());
+			return packageFragmentRoot.getJavaProject().findPackageFragment(currentFolder.getFullPath());
 		}
 		
-		return null;
+		throw new CoreException(new Status(IStatus.ERROR, JBossJaxrsUIPlugin.PLUGIN_ID, "Cannot retrieve the currentFolder for the given packageFragmentRoot."));
```

Comment:
Maybe having the `currentFolder` and `packageFragmentRoot` in the log message may help debugging

Reply:
at this point, currentFolder is null ( that's what's causing the NPE) so I'll only add the `packageFragmentRoot ` in the log